Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.py

```diff
@@ -0,0 +1,1078 @@
+            message = 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return message
+
+# region Methods
+# ---------------
+
+
+def securityscorecard_portfolios_list_command(client: Client) -> CommandResults:
+    """`securityscorecard_portfolios_list_command`: List all Portfolios you have access to.
+
+    See https://securityscorecard.readme.io/reference#get_portfolios
+
+    :type ``client``: ``Client``
+
+    :return:
+        A ``CommandResults`` object that is passed to ``return_results``
+    :rtype: ``CommandResults``
+    """
+
+    portfolios = client.get_portfolios()
+
+    # For mypy
+    # error: Argument 1 to "int" has incompatible type "Optional[Any]";
+    # expected "Union[str, bytes, SupportsInt, _SupportsIndex]"
+    portfolios_total = portfolios.get("total")
+    assert portfolios_total is not None
+    portfolios_count = int(portfolios_total)
+
+    # Check that API returned more than 0 portfolios
+    if portfolios_count == 0:
+        return_warning("No Portfolios were found in your account. Please create a new one and try again.", exit=True)
+
+    # API response is a dict with 'entries'
+    entries = portfolios.get('entries')
+
+    markdown = tableToMarkdown('Your SecurityScorecard Portfolios', entries, headers=['id', 'name', 'privacy'])
+
+    results = CommandResults(
+        readable_output=markdown,
+        outputs_prefix='SecurityScorecard.Portfolio',
+        outputs_key_field='id',
+        outputs=portfolios
+    )
+
+    return results
+
+
+def securityscorecard_portfolio_list_companies_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """`securityscorecard_portfolio_list_companies_command`: Retrieve all companies in portfolio.
+
+    https://securityscorecard.readme.io/reference#get_portfolios-portfolio-id-companies
+
+    :type ``client``: ``Client``
+    :type `` args``: ``Dict[str, Any]``
+        ``args['portfolio_id']``: Portfolio ID.
+            A list of Portfolio IDs can be retrieved using the `!securityscorecard-portfolios-list` command., type ``String``
+        ``args['grade']``: Grade filter. The acceptable values are capitalized letters between A-F, e.g. B., type ``String``
+        ``args['industry']``: Industry filter.
+            The acceptable values are capitalized, e.g. INFORMATION_SERVICES, TECHNOLOGY., type ``String``
+        ``args['vulnerability']``: Vulnerability filter, type ``String``
+        ``args['issue_type']``: Issue type filter, type ``String``
+        ``args['had_breach_within_last_days']``:
+            Domains with breaches in the last X days. Possible values are numbers, e.g. 1000., type ``Number``
+
+    :return:
+        A ``CommandResults`` object that is passed to ``return_results``
+    :rtype: ``CommandResults``
+    """
+
+    demisto.debug("securityscorecard_portfolio_list_companies_command called with args: {0}".format(args))
+
+    portfolio_id = str(args.get('portfolio_id'))
```

Comment:
I actually had it originally at:
```python
portfolio_id = args.get('portfolio_id')
```

But when I ran validation, it threw a compatibility error:

```bash
mypy SecurityScorecard.py
SecurityScorecard.py:463: error: Argument "portfolio" to "get_companies_in_portfolio" of "Client" has incompatible type "Optional[Any]"; expected "str"
```

This is because `Client::get_companies_in_portfolio` expects a `str` for the `portfolio` argument:

```python
def get_companies_in_portfolio(
        self,
        portfolio: str,
        ...
)
```

The problem is true to all other instances where I cast an argument to `str`.

What do you suggest to do in this case?

Reply:
I think setting the type of `args` to be dict of string should fix it -`Dict['str',str']`. If not sit with Tom he should be able to help. Let me know if he's able to help.